lib/Dancer2/Core/App.pm

```diff
@@ -889,7 +889,11 @@ sub send_file {
                 [ $res->status, $res->headers_to_array, $fh ]
             );
         };
+
+        Scalar::Util::weaken( my $weak_self = $self );
+
         $response = Dancer2::Core::Response::Delayed->new(
+            error_cb => sub { $weak_self->logger_engine->log(@_) },
```

Comment:
Sorry to go MIA on this. I'm sure its missing an arg.  
  * `error_cb` takes the error message as its only argument. 
  * The `log` method on all the logging engines take two args: "msg_level", and  "msg_content".

The error callback added to the `content` keyword in this PR  [two args for its logging call](https://github.com/PerlDancer/Dancer2/blob/feature/delayed-on-error/lib/Dancer2/Core/DSL.pm#L301-L303)

This is inside of `send_file`, which uses delayed responses to send a filehandle back to the PSGI engine. So `$Dancer2::Core::Route::WRITER->write` shouldn't be called in this use case; I'm not certain its possible to write a test case for it (at least I can't think of how at the moment).
 [ The fix would be: `error_cb => sub { $weak_self->logger_engine->log(warning => @_) }` ]